src/site/content/en/blog/pwas-on-oculus-2/index.md

```diff
@@ -296,66 +296,136 @@ customize the PWA experience.
+
+[Bubblewrap](https://github.com/GoogleChromeLabs/bubblewrap) is an open source set of libraries and
+a command line tool (CLI) for Node.js developed by the Google Chrome team that helps developers
+generate, build, and sign an Android project that launches your PWA as a
+[Trusted Web Activity](https://developer.chrome.com/docs/android/trusted-web-activity/) (TWA).
+
+Meta Quest Browser currently doesn't fully support TWA, but starting from version 1.18.0,
+Bubblewrap [supports](https://github.com/GoogleChromeLabs/bubblewrap/releases/tag/v1.18.0) packaging
+PWAs for Meta Quest devices.
+
+It can generate *universal* APKs that open a TWA on regular Android devices and open Meta Quest
+Browser on Meta Quest devices, so it can be published to the Google Play Store and the Meta Quest Store.
+
+Assuming you have [Node.js](https://nodejs.org) installed, Bubblewrap CLI can be installed with the
+following command:
+
+```bash
+npm i -g @bubblewrap/cli
+```
+
+When running Bubblewrap for the first time, it will offer to automatically download and install the
+required external dependencies—Java Development Kit (JDK) and Android SDK Build Tools.
+
+To generate a Meta Quest compatible Android project that wraps your PWA, run the `init` command
+with the `--metaquest` flag and follow the wizard:
+
+```bash
+bubblewrap init --manifest="https://your.web.app/manifest.json" --metaquest
+```
+
+Once the project has been generated, build and sign it by running:
+
+```bash
+bubblewrap build
+```
+
+This will output a file called `app-release-signed.apk`. This file can be installed on the device
+or uploaded to the Meta Quest Store and the Google Play Store.
+
+{% Aside 'important' %} You can also use the `--chromeosonly` flag in addition to the `--metaquest`
+flag to make the APK compatible not only with Meta Quest and regular Android devices, but also with
+[ChromeOS](https://chromeos.dev) devices. {% endAside %}
+
+### Packaging PWAs with Oculus Platform Utility
+
+[Oculus Platform Utility](https://developer.oculus.com/documentation/web/pwa-packaging/#download-the-cli)
+is the official command line tool developed by Meta for publishing apps for Oculus Rift and Meta
+Quest devices.
+
+It also allows to package PWAs for Meta Quest devices with the `create-pwa` command and publish
+them to Meta Quest Store and App Lab.
+
+Set the output file name via the `-o` parameter and the path to Android SDK via the `--android-sdk`
+parameter.
+
+Point the tool at the live URL of the web app manifest via the `--web-manifest-url` parameter.
+
+If you don't have a manifest on your live PWA or wish to override the live manifest, you can still
+generate an APK for your PWA using a local manifest file and the `--manifest-content-file`
+parameter.
+
+To leave the manifest as pure as possible, use the `--package-name` parameter with a value in
+reverse domain name notation (e.g., `com.company.app.pwa`), rather than adding the proprietary
+`ovr_package_name` field to the manifest.
```

Comment:
```suggestion
To leave the manifest as pure as possible, use the `--package-name` parameter with a value in
reverse domain name notation (for example, `com.company.app.pwa`), rather than adding the proprietary
`ovr_package_name` field to the manifest.
```